### PR TITLE
#797 fix collections iterator

### PR DESCRIPTION
**Pull request by asolntsev**
implementation of https://github.com/selenide/selenide/issues/797

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/ElementsCollection.java [hunk 1]
```diff
@@ -443,43 +444,64 @@ public int size() {
     }
   }
 
+  /**
+   * @deprecated I don't think this method is needed in tests.
```
[the remark this answers is not in the document]
Comment (bereg2k) [reply src/main/java/com/codeborne/selenide/ElementsCollection.java:448]: Also, why override these methods anyway? Simply for discouraging the usage of them via documentation? =)

File: src/test/java/integration/CollectionWithChangingTextsTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    assertThat(it.hasNext()).isFalse();
+  }
+
+  @Test
+  void reloadingIterable_reloadsCollectionOnEveryCall() throws InterruptedException {
+    Iterator<SelenideElement> it = $$("#collection li").asReloadingIterable().iterator();
+    it.next().shouldHave(text("Item #1"));
+    it.next().shouldHave(text("Item #2"));
+    it.next().shouldHave(text("Item #3"));
+    sleep(350);
```
Comment (bereg2k) [inline src/test/java/integration/CollectionWithChangingTextsTest.java:47]: IMO, this looks a bit "magical", implicit.

Maybe it's better to leave a comment here that the page has a script that adds additional elements to the collection after this timeout?

File: src/main/java/com/codeborne/selenide/ElementsCollection.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/ElementsCollection.java:448]: Yes, I wanted to override these methods just to mark them as deprecated. 
So people who occasionally used these methods will see a warning in IDE.

File: src/test/java/integration/CollectionWithChangingTextsTest.java (hunk 2 above)
Comment (asolntsev) [reply src/test/java/integration/CollectionWithChangingTextsTest.java:47]: yes, I will add a comment. 
Though, many of Selenide tests use such approach: an html page with dynamic JS inside and _a Java test.

**Pull request merged**